torch_xla/csrc/helpers.cpp

```diff
@@ -578,6 +700,137 @@ std::pair<xla::XlaOp, xla::XlaOp> XlaHelpers::PromoteSecond(xla::XlaOp op1,
   return PromoteShapes(vops.first, vops.second);
 }
 
+#if EXPERIMENTAL_XLA_UNBOUNDED_DYNAMISM
+xla::XlaOp XlaHelpers::ImplicitBroadcastWithUnboundedDynamicShapes(
+    xla::XlaOp op, const xla::Shape& op_shape, xla::XlaOp aux_op,
+    const xla::Shape& shape) {
+  XLA_CHECK(XlaHelpers::IsUnboundedDynamic(shape) ||
+            XlaHelpers::IsUnboundedDynamic(op_shape));
+
+  const xla::Shape& aux_shape = ShapeHelper::ShapeOfXlaOp(aux_op);
+  const auto& op_shape_dims = op_shape.dimensions();
+  const auto& aux_shape_dims = aux_shape.dimensions();
+  const auto& shape_dims = shape.dimensions();
+
+  XLA_CHECK_GE(shape_dims.size(), op_shape_dims.size())
+      << shape << " vs " << op_shape;
+
+  int64_t size_delta = shape_dims.size() - op_shape_dims.size();
+  xla::XlaOp new_op = op;
+  std::vector<xla::XlaOp> get_dim_ops;
+  std::vector<xla::XlaOp> reshaped_ops;
+
+  if (size_delta > 0) {
+    std::cout << "\t size_delta > 0\n";
+    std::vector<int64_t> broadcast_sizes(shape_dims.begin(),
+                                         shape_dims.begin() + size_delta);
+    for (int i = 0; i < size_delta; i++) {
+      if (broadcast_sizes[i] != kUnboundedSize) {
+        get_dim_ops.push_back(
+            XlaHelpers::ScalarValue<int32_t>(broadcast_sizes[i], op.builder()));
+
+        auto s = ShapeHelper::ShapeOfXlaOp(get_dim_ops.back());
+        std::cout << "implicitB shape: " << xla::ShapeUtil::HumanString(s)
+                  << " for size: " << broadcast_sizes[i] << "\n";
```

Comment:
we should use TF_VLOG instead of cout

Reply:
Let me remove all of these debugging prints